### PR TITLE
Add overmapgen coverage test, and fix related issues

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
Bugfixes "Fix various overmap special and city building spawns"

#### Purpose of change
Prior to this PR we had no good way to detect when certain overmap things which were supposed to spawn didn't.  Such failure to spawn could be for various reasons, for example failing to add a city building to regional settings.

#### Describe the solution
Add a test that generates a bunch of overmaps and looks to see what was placed there.  Then compare that with the list of all `oter_type_id`s and see what's missing.

Add a new flag `SHOULD_NOT_SPAWN` for overmap terrains which should not be expected to spawn.  Apply it to various terrains for which this is true.  Mostly test stuff, and works in progress (like the campus map).

Remove several overmap specials, terrains, mapgen for obsolete things that haven't spawned in a long time.  As far as I know, we don't have migration support for overmap terrains (not in a form that would be particularly useful for this, anyway).  I think these have all not spawned since at least 2019 so I think they should be relatively safe to remove.  If people start getting errors feel free to let me know and we can reinstate any as appropriate.

Removed things were:
* `apartments_con` (superseded by `apartments_con_new`)
* `apartments_mod` (superseded by `apartments_mod_new`)
* `dirtroad` (superseded by `dirt_road`)
* Various farm OMTs for specific lot layouts that it seems weren't used but were added for completeness.  I could have marked them `SHOULD_NOT_SPAWN`, but I figured they'd be easy enough to recreate if needed.
* `home_improvement_superstore_new` (superseded by `home_improvement_superstore_new`).
* `microlab_generic_isolated_stairs_even`
* `4way_road` (I think superseded by linear terrain)
* `looted_building`
* `temple` (superseded by `temple_stairs`)
* `cavern`

Fix a bunch of other issues causing things to not spawn:
* Mansion terrains had unused alias names which appeared to exist purely for organizational purposes.
* Add a missing special for `natural spring`.
* Add a missing special for `campground_npc` (looks like this was accidentally deleted in a merge resolution at some point).
* Add missing city building entry for `used_bookstore`.
* There were two specials named `Military Bunker` and two named `Military Outpost`.  Rename one of each so they don't overwrite the others.
* Fix a typo in the central lab special.

A few things spawn sometimes but not reliably, so they got special treatment:
* Ant labs are very unlikely to spawn (ants and a lab have to randomly spawn in the same place).
* Marinas usually spawn, but fail often enough that they'd have made the test flaky.
* Ravines only spawn when either the desert region or Aftershock is active.
* `s_restaurant_deserted_test` only spawns in the desert region.
* `rock_border` only spawns in its special scenario.

#### Describe alternatives you've considered
A lot of subjective decisions between marking something `SHOULD_NOT_SPAWN` vs just deleting it.  I chose to delete when I thought that (a) it was not likely to ever be turned into something useful and (b) it was old enough that it probably wouldn't appear in people's ongoing games.  But the first is subjective and the second is sometimes hard to determine in our messy version history, so I may have made a mistake.

The military bunker and outpost which weren't spawning before look somewhat boring, so maybe they would have been better removed, but I didn't look in detail.

#### Testing
I soak tested this unit test for about seven hours without a failure, so I'm fairly confident it's sufficiently non-flaky.

#### Additional context
Some folks who may want to check my decisions: @curstwist @John-Candlebury @LovamKicsiGazsii.

### CODE REVIEW

**Comment by jbytheway** [comment]
Good to know.  I'll convert to draft while I look into that.

**Comment by jbytheway** [comment]
I was able to start a defense mode game in the Mansion without any problems.  And I don't see anything in the code that refers to the OMTs I removed, so I think we're good.  Marked ready for review again.

**Comment by kevingranade** [comment]
Graphical Overmap and alt map key mods have dangling references, They probably just need to be removed but I don't have time to jump in at the moment.

**Comment by jbytheway** [comment]
I've pushed a change with my best effort to remove the relevant pieces from the mods.

File: doc/JSON_FLAGS.md [hunk 1]
```diff
@@ -1199,6 +1199,9 @@ These branches are also the valid entries for the categories of `dreams` in `dre
 - ```KNOWN_UP``` There's a known way up.
 - ```LINEAR``` For roads etc, which use ID_straight, ID_curved, ID_tee, ID_four_way.
 - ```NO_ROTATE``` The terrain can't be rotated (ID_north, ID_east, ID_south, and ID_west instances will NOT be generated, just ID).
+- ```SHOULD_NOT_SPAWN``` The terrain should not be expected to spawn.  This
+  might be because it exists only for testing purposes, or it is part of a
+  partially completed feature wher emore work is required before it can start spawning.
```
Comment (Night-Pryanik) [inline doc/JSON_FLAGS.md:1204]:
```suggestion
  partially completed feature where more work is required before it can start spawning.
```

**Review by Night-Pryanik — APPROVED** [review]

**Pull request merged**